soc/arm64/intel_socfpga/agilex/soc.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2021, Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _SOC_H_
+#define _SOC_H_
+
+#include <arch/cpu.h>
+
+/* FPGA config helpers */
+#define INTEL_SIP_SMC_FPGA_CONFIG_ADDR		0x400000
+#define INTEL_SIP_SMC_FPGA_CONFIG_SIZE		0x2000000
+
+/* Register Mapping */
+#define SOCFPGA_CCU_NOC_REG_BASE		0xf7000000
+
+#define SOCFPGA_MMC_REG_BASE			0xff808000
+
+#define SOCFPGA_RSTMGR_REG_BASE			0xffd11000
+#define SOCFPGA_SYSMGR_REG_BASE			0xffd12000
+
+#define SOCFPGA_L4_PER_SCR_REG_BASE             0xffd21000
+#define SOCFPGA_L4_SYS_SCR_REG_BASE             0xffd21100
+#define SOCFPGA_SOC2FPGA_SCR_REG_BASE           0xffd21200
+#define SOCFPGA_LWSOC2FPGA_SCR_REG_BASE         0xffd21300
+
+#define mmio_write_32(addr, data) \
+	sys_write32((data), (addr))
+
+#define mmio_read_32(addr) \
+	sys_read32((addr))
+
+#define mmio_setbits_32(addr, bit) \
+	sys_write32(sys_read32((addr)) | (bit), (addr))
+
+#define mmio_clrbits_32(addr, bit) \
+	sys_write32(sys_read32((addr)) & ~(bit), (addr))
```

Comment:
Use `sys_set_bit` and `sys_clear_bit` instead.

Reply:
Hi Carlo,

In our use case, the 'bit' represent mask, our code will set/clear more that 1 bits using this command. 
I will change bit to mask in the macro to confusion.
    mmio_setbits_32(addr, bit) => mmio_setbits_32(addr, mask)

sys_set_bit and sys_clear_bit inline function accept bit offset. It does not fit our function. To centralize bitops in zephyr, is it possible to provide the macro that set/clear bits using mask?
    Example: sys_set_bits (addr, mask)

static ALWAYS_INLINE void sys_set_bit(mem_addr_t addr, unsigned int bit)
{
	uint32_t temp = *(volatile uint32_t *)addr;

	*(volatile uint32_t *)addr = temp | (1 << bit);
}

Thanks,
Siew Chin

Reply:
> is it possible to provide the macro that set/clear bits using mask?

Feel free to add the new macros you need in `include/arch/common/sys_bitops.h`

Reply:
Noted.